### PR TITLE
tests: add test benchmark script

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: tests/util/benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (zyga) [inline tests/util/benchmark.sh]: If you don't need any bash-specific extensions using /bin/sh is probably more desirable.

File: tests/util/benchmark.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+BACKEND=${1:-"linode:"}
```
Comment (zyga) [inline tests/util/benchmark.sh]: I think quotes should be like this instead:

`BACKEND="${1:-linode}"`

File: tests/util/benchmark.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ITERATIONS=${2:-10}
+OUTPUT_FILE=${3:-"${PWD}/benchmark.out"}
```
Comment (zyga) [inline tests/util/benchmark.sh]: Same, if the third argument needs quoting this will not expand correctly here.

File: tests/util/benchmark.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+SUCCESSFUL_EXECUTIONS=0
+
+rm -rf "$OUTPUT_FILE"
```
Comment (zyga) [inline tests/util/benchmark.sh]: If `$OUTPUT_FILE` is a file then plain `rm -f` is probably safer

File: tests/util/benchmark.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+for i in $(seq 1 "$ITERATIONS"); do
```
Comment (zyga) [inline tests/util/benchmark.sh]: From a quick check I think that `seq` is already 1-based and you don't need the extra argument

File: tests/util/benchmark.sh (hunk 1 above)
Comment (fgimenez) [inline tests/util/benchmark.sh]: Yes, this comes from a shellcheck suggestion, using the `$SECONDS` bashism requires it.

File: tests/util/benchmark.sh (hunk 2 above)
Comment (fgimenez) [inline tests/util/benchmark.sh]: Sure, done

File: tests/util/benchmark.sh (hunk 3 above)
Comment (fgimenez) [inline tests/util/benchmark.sh]: It seemed to work, the .out files were written correctly in the current directory, but done anyway for consistency.

File: tests/util/benchmark.sh (hunk 4 above)
Comment (fgimenez) [inline tests/util/benchmark.sh]: Good catch, done

File: tests/util/benchmark.sh (hunk 5 above)
Comment (fgimenez) [inline tests/util/benchmark.sh]: Didn't know, thanks :) done

**Comment by fgimenez** [comment]
hey @zyga thanks a lot for your review! :) And thanks for the suggestion about `shellcheck`, i have it integrated with flycheck and find it really useful :)

File: tests/util/benchmark.sh (hunk 1 above)
Comment (niemeyer) [inline tests/util/benchmark.sh]: @zyga spread uses bash exclusively, as a much saner default.

**Pull request merged**